File: GameLibrary/src/cc/lib/swing/AWTApplet.java
package cc.lib.swing;

import java.net.MalformedURLException;
import java.net.URL;
import java.util.Collection;
import java.util.List;

import javax.swing.JApplet;

import cc.lib.utils.Reflector;

public abstract class AWTApplet extends JApplet {

    @Override
    public final void init() {
        Reflector.DISABLED = true;
        AWTImageMgr.applet = this;
        initApp();
    }

    public abstract URL getAbsoluteURL(String imagePath) throws MalformedURLException;

    protected abstract void initApp();

    public <T extends Enum<T>> List<T> getEnumListProperty(String property, Class enumClass, List<T> defaultList) {
        return defaultList;
    }

    public String getStringProperty(String property, String defaultValue) {
        return defaultValue;
    }

    public void setStringProperty(String s, String v) {}

    public <T extends Enum<T>> void setEnumListProperty(String s, Collection<T> l) {}

    public void setIntProperty(String s, int value) {}

    public int getIntProperty(String s, int defaultValue) { return defaultValue; }

}

File: SwingApps/src/cc/lib/swing/AWTApplet.kt
package cc.lib.swing

import cc.lib.reflector.Reflector
import java.net.MalformedURLException
import java.net.URL
import javax.swing.JApplet

abstract class AWTApplet : JApplet() {
	override fun init() {
		Reflector.DISABLED = true
		AWTImageMgr.applet = this
		initApp()
	}

	@Throws(MalformedURLException::class)
	abstract fun getAbsoluteURL(imagePath: String): URL
	protected abstract fun initApp()
	open fun <T : Enum<T>> getEnumListProperty(property: String, clazz: Class<T>, defaultList: List<T>): List<T> {
		return defaultList
	}

	open fun getStringProperty(property: String, defaultValue: String): String {
		return defaultValue
	}

	open fun setStringProperty(s: String, v: String) {}
	open fun <T : Enum<T>> setEnumListProperty(s: String, l: Collection<T>) {}
	open fun setIntProperty(s: String, value: Int) {}
	open fun getIntProperty(s: String, defaultValue: Int): Int {
		return defaultValue
	}

	open fun <T : Enum<T>> getEnumProperty(s: String, clazz: Class<T>, defaultValue: T): T {
		return defaultValue
	}

	open fun <T : Enum<T>> setEnumProperty(s: String, value: T) {}
}